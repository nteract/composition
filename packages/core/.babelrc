{
  "env": {
    "test": {
      "presets": ["es2015", "react"],
      "plugins": [
        "styled-jsx/babel",
        "transform-class-properties",
        "transform-flow-strip-types",
        "transform-object-rest-spread"
      ]
    }
  },
  "presets": ["es2015", "react"],
  "plugins": [
    "styled-jsx/babel",
    "transform-class-properties",
    "transform-flow-strip-types",
    "transform-object-rest-spread"
  ]
}
