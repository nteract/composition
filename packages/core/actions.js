// @flow
module.exports = require("./").actions;
