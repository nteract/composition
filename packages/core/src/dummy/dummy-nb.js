// @flow
/* eslint-disable no-multi-str */
import { fromJS } from "@nteract/commutable";

export const dummy =
  '{"cells":[{"cell_type":"markdown","metadata":{},\
"source":["## The Notable Nteract Notebook\\n","\\n","**It\'s a notebook!**\\n"]},\
{"cell_type":"code","execution_count":11,"metadata":{"collapsed":false},\
"outputs":[{"data":{"text/plain":["<h1>Multiple</h1>"],\
"text/plain":["<IPython.core.display.HTML object>"]},"metadata":{},"output_type":"display_data"}],\
"source":["import IPython\\n","\\n","from IPython.display import HTML\\n",\
"from IPython.display import Markdown\\n","from IPython.display import display\\n","\\n",\
"display(HTML(\'<h1>Multiple</h1>\'))\\n","display(HTML(\'<p>Display Elements</p>\'))\\n",\
"display(Markdown(\'**awesome**\'))\\n","\\n","print(\'hey\')\\n","42"]}],\
"metadata":{"kernelspec":{"display_name":"Python 3","language":"python","name":"python3"},\
"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py",\
"mimetype":"text/x-python","name":"python","nbconvert_exporter":"python",\
"pygments_lexer":"ipython3","version":"3.5.1"}},"nbformat":4,"nbformat_minor":0}';

export const dummyJSON = JSON.parse(dummy);

export const dummyCommutable = fromJS(dummyJSON);

export const bigDummyJSON = {
  cells: [
    {
      cell_type: "markdown",
      source: [
        "# This is an example notebook\n",
        "\n",
        "## Used for test purposes...\n",
        "\nThe idea is to have enough real data in here to make a non-trivial NB."
      ],
      metadata: {
        trusted: false
      }
    },
    {
      cell_type: "code",
      source: ["import math\n", "import IPython.display"],
      outputs: [],
      execution_count: 17,
      metadata: {
        trusted: false
      }
    },
    {
      cell_type: "markdown",
      source: [
        "Here, we intersperse markdown to make sure it isn't simply a bunch of consecutive code blocks."
      ],
      metadata: {
        collapsed: false,
        outputHidden: false,
        inputHidden: false
      }
    },
    {
      cell_type: "code",
      source: [
        "data = [round(math.sin(i*2*math.pi/10)*100) + 100 for i in range(0, 10)]"
      ],
      outputs: [],
      execution_count: 23,
      metadata: {
        collapsed: false,
        outputHidden: false,
        inputHidden: false
      }
    },
    {
      cell_type: "markdown",
      source: [
        "Finally, we even display some HTML in the output to ensure that we have more than simple text output."
      ],
      metadata: {}
    },
    {
      cell_type: "code",
      source: [
        'bars = "".join([\'<div style="display: inline-block; margin: 5px; background-color: #0074D9; width: 20px; height: {}px;"></div>\'.format(datum) for datum in data])\n',
        "IPython.display.HTML(bars)"
      ],
      outputs: [
        {
          output_type: "execute_result",
          execution_count: 33,
          data: {
            "text/plain": ["<IPython.core.display.HTML object>"],
            "text/html": [
              '<div style="display: inline-block; margin: 5px; background-color: #0074D9; width: 20px; height: 100px;"></div><div style="display: inline-block; margin: 5px; background-color: #0074D9; width: 20px; height: 159px;"></div><div style="display: inline-block; margin: 5px; background-color: #0074D9; width: 20px; height: 195px;"></div><div style="display: inline-block; margin: 5px; background-color: #0074D9; width: 20px; height: 195px;"></div><div style="display: inline-block; margin: 5px; background-color: #0074D9; width: 20px; height: 159px;"></div><div style="display: inline-block; margin: 5px; background-color: #0074D9; width: 20px; height: 100px;"></div><div style="display: inline-block; margin: 5px; background-color: #0074D9; width: 20px; height: 41px;"></div><div style="display: inline-block; margin: 5px; background-color: #0074D9; width: 20px; height: 5px;"></div><div style="display: inline-block; margin: 5px; background-color: #0074D9; width: 20px; height: 5px;"></div><div style="display: inline-block; margin: 5px; background-color: #0074D9; width: 20px; height: 41px;"></div>'
            ]
          },
          metadata: {}
        }
      ],
      execution_count: 33,
      metadata: {
        collapsed: false,
        outputHidden: false,
        inputHidden: false
      }
    },
    {
      cell_type: "code",
      source: [],
      outputs: [],
      execution_count: 28,
      metadata: {
        collapsed: false,
        outputHidden: false,
        inputHidden: false
      }
    },
    {
      cell_type: "markdown",
      source: [
        "^^ Add an empty code cell above and an empty markdown cell below for kicks. Et, voila."
      ],
      metadata: {}
    },
    {
      cell_type: "markdown",
      source: [],
      metadata: {}
    }
  ],
  metadata: {
    kernelspec: {
      display_name: "Python 3",
      language: "python",
      name: "python3"
    },
    language_info: {
      name: "python",
      version: "3.6.0",
      mimetype: "text/x-python",
      codemirror_mode: {
        name: "ipython",
        version: 3
      },
      pygments_lexer: "ipython3",
      nbconvert_exporter: "python",
      file_extension: ".py"
    }
  },
  nbformat: 4,
  nbformat_minor: 2
};

export const bigDummyCommutable = fromJS(bigDummyJSON);

export const bigDummy = JSON.stringify(bigDummyJSON);
