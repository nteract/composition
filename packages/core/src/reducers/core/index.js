// @flow
import { combineReducers } from "redux-immutable";

import * as actionTypes from "../../actionTypes";
import { makeStateRecord } from "../../state";

import { communication } from "./communication";
import { entities } from "./entities";

// TODO: #2618: This should at a minimum be moved into a contents entry.
// TODO: This is temporary until we have sessions in place. Ideally, we point to
// a document, which knows about its session and that session knows about its
// kernel. For now, we need to keep a reference to the currently targeted kernel
// around.
const kernelRef = (state = null, action) => {
  switch (action.type) {
    case actionTypes.LAUNCH_KERNEL:
    case actionTypes.LAUNCH_KERNEL_BY_NAME:
      return action.payload.selectNextKernel ? action.payload.kernelRef : state;
    case actionTypes.LAUNCH_KERNEL_SUCCESSFUL:
      return action.payload.selectNextKernel ? action.payload.kernelRef : state;
    default:
      return state;
  }
};

const currentKernelspecsRef = (state = null, action) => {
  switch (action.type) {
    case actionTypes.FETCH_KERNELSPECS:
      return action.payload.kernelspecsRef;
    default:
      return state;
  }
};

const core = combineReducers(
  {
    communication,
    currentKernelspecsRef,
    entities,
    kernelRef
  },
  makeStateRecord
);

export default core;
