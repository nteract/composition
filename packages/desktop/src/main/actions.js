export function setKernelSpecs(kernelSpecs: Object) {
  return {
    type: "SET_KERNELSPECS",
    kernelSpecs: kernelSpecs
  };
}
