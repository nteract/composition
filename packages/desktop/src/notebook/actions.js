// @flow
import * as constants from "./constants";

import type {
  ImmutableCell,
  ImmutableNotebook,
  CellID,
  CellType,
  ImmutableCellOrder,
  ImmutableOutput,
  ImmutableOutputs,
  MimeBundle
} from "@nteract/commutable/src/types";

export function newKernel(kernelSpec: any, cwd: string) {
  return {
    type: constants.LAUNCH_KERNEL,
    kernelSpec,
    cwd
  };
}

export function newKernelByName(kernelSpecName: any, cwd: string) {
  return {
    type: constants.LAUNCH_KERNEL_BY_NAME,
    kernelSpecName,
    cwd
  };
}

export function setNotebookKernelInfo(kernelInfo: any) {
  return {
    type: constants.SET_KERNEL_INFO,
    kernelInfo
  };
}

// TODO: Lock this type down to a proper enum
export function setExecutionState(executionState: string) {
  return {
    type: constants.SET_EXECUTION_STATE,
    executionState
  };
}

export function clearOutputs(id: string) {
  return {
    type: constants.CLEAR_OUTPUTS,
    id
  };
}

// TODO: above doesn't appear to get used for a MoveCellAction
export function moveCell(id: string, destinationId: string, above: boolean) {
  return {
    type: constants.MOVE_CELL,
    id,
    destinationId,
    above
  };
}

export function removeCell(id: string) {
  return {
    type: constants.REMOVE_CELL,
    id
  };
}

export function createCellAfter(
  cellType: CellType,
  id: string,
  source: string = ""
) {
  return {
    type: constants.NEW_CELL_AFTER,
    source,
    cellType,
    id
  };
}

export function createCellBefore(cellType: CellType, id: string) {
  return {
    type: constants.NEW_CELL_BEFORE,
    cellType,
    id
  };
}

export function createCellAppend(cellType: CellType) {
  return {
    type: constants.NEW_CELL_APPEND,
    cellType
  };
}

export function mergeCellAfter(id: string) {
  return {
    type: constants.MERGE_CELL_AFTER,
    id
  };
}

/**
 * setInCell can generically be used to set any attribute on a cell, including
 * and especially for changing metadata per cell.
 * @param {CellID} id    cell ID
 * @param {Array<string>} path  path within a cell to set
 * @param {any} value what to set it to
 *
 * Example:
 *
 * > action = setInCell('123', ['metadata', 'cool'], true)
 * > documentReducer(state, action)
 * {
 *   ...
 *   '123': {
 *     'metadata': {
 *       'cool': true
 *     }
 *   }
 * }
 *
 */
export function setInCell(id: CellID, path: Array<string>, value: any) {
  return {
    type: "SET_IN_CELL",
    id,
    path,
    value
  };
}

export function updateCellSource(id: string, source: string) {
  return setInCell(id, ["source"], source);
}

export function updateCellExecutionCount(id: string, count: number) {
  return setInCell(id, ["execution_count"], count);
}

export function changeOutputVisibility(id: string) {
  return {
    type: constants.CHANGE_OUTPUT_VISIBILITY,
    id
  };
}

export function changeInputVisibility(id: string) {
  return {
    type: constants.CHANGE_INPUT_VISIBILITY,
    id
  };
}

// TODO de-anyify this signature
export function updateCellPagers(id: string, pagers: any) {
  return {
    type: constants.UPDATE_CELL_PAGERS,
    id,
    pagers
  };
}

// TODO de-anyify this signature
export function updateCellStatus(id: string, status: any) {
  return {
    type: constants.UPDATE_CELL_STATUS,
    id,
    status
  };
}

// TODO de-anyify this signature
export function focusCell(id: string) {
  return {
    type: constants.FOCUS_CELL,
    id
  };
}

export function focusNextCell(id: string, createCellIfUndefined: boolean) {
  return {
    type: constants.FOCUS_NEXT_CELL,
    id,
    createCellIfUndefined
  };
}

export function focusPreviousCell(id: string) {
  return {
    type: constants.FOCUS_PREVIOUS_CELL,
    id
  };
}

export function focusCellEditor(id: string) {
  return {
    type: constants.FOCUS_CELL_EDITOR,
    id
  };
}

export function focusNextCellEditor(id: string) {
  return {
    type: constants.FOCUS_NEXT_CELL_EDITOR,
    id
  };
}

export function focusPreviousCellEditor(id: string) {
  return {
    type: constants.FOCUS_PREVIOUS_CELL_EDITOR,
    id
  };
}

export function toggleStickyCell(id: string) {
  return {
    type: constants.TOGGLE_STICKY_CELL,
    id
  };
}

export function overwriteMetadata(field: string, value: any) {
  return {
    type: constants.OVERWRITE_METADATA_FIELD,
    field,
    value
  };
}

export function deleteMetadata(field: string) {
  return {
    type: constants.DELETE_METADATA_FIELD,
    field
  };
}

export const killKernel = {
  type: constants.KILL_KERNEL
};

export const interruptKernel = {
  type: constants.INTERRUPT_KERNEL
};

export function setNotificationSystem(notificationSystem: any) {
  return {
    type: constants.SET_NOTIFICATION_SYSTEM,
    notificationSystem
  };
}

export function copyCell(id: CellID) {
  return {
    type: constants.COPY_CELL,
    id
  };
}

export function cutCell(id: CellID) {
  return {
    type: constants.CUT_CELL,
    id
  };
}

export function pasteCell() {
  return {
    type: constants.PASTE_CELL
  };
}

export function changeCellType(id: CellID, to: CellType) {
  return {
    type: constants.CHANGE_CELL_TYPE,
    id,
    to
  };
}

export function setGithubToken(githubToken: string) {
  return {
    type: constants.SET_GITHUB_TOKEN,
    githubToken
  };
}

export function setConfigKey(key: string, value: any) {
  return {
    type: constants.SET_CONFIG_KEY,
    key,
    value
  };
}

export function setTheme(theme: string) {
  return setConfigKey("theme", (theme: string));
}

export function setCursorBlink(value: string) {
  return setConfigKey("cursorBlinkRate", value);
}

export function toggleOutputExpansion(id: string) {
  return {
    type: constants.TOGGLE_OUTPUT_EXPANSION,
    id
  };
}

/**
 * Execute Cell action.
 *
 * @param {String} id - Universally Unique Identifier of cell to be executed.
 * @param {Object} source - Source code to executed.
 * @return {Object} executeCellAction - Action to be dispatched to reducer.
 */
export function executeCell(id: string, source: string) {
  return {
    type: constants.EXECUTE_CELL,
    id,
    source
  };
}

export function changeFilename(filename: string) {
  return {
    type: constants.CHANGE_FILENAME,
    filename
  };
}

export function save(filename: string, notebook: any) {
  return {
    type: constants.SAVE,
    filename,
    notebook
  };
}

export function saveAs(filename: string, notebook: any) {
  return {
    type: constants.SAVE_AS,
    filename,
    notebook
  };
}

export function doneSaving(notebook: any) {
  return {
    type: constants.DONE_SAVING,
    notebook
  };
}
