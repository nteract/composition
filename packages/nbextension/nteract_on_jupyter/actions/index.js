// @flow
import type { CONTENTS_ACTIONS } from "./contents";
import type { KERNELSPECS_ACTIONS } from "./kernelspecs";

export type GENERIC_AJAX_FAIL = {
  type: "GENERIC_AJAX_FAIL",
  payload: any,
  status: number
};

export function genericAjaxFail(ajaxError: any): GENERIC_AJAX_FAIL {
  return {
    type: "GENERIC_AJAX_FAIL",
    payload: ajaxError.response,
    status: ajaxError.status
  };
}

export type ALL_ACTIONS =
  | CONTENTS_ACTIONS
  | KERNELSPECS_ACTIONS
  | GENERIC_AJAX_FAIL;
