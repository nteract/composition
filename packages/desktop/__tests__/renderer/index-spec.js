import App from "./../../src/notebook/index.js";
import * as actions from "@nteract/core/actions";

import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { ipcRenderer as ipc } from "electron";

import { dummyCommutable, dummyJSON } from "dummy-nb";

describe("App", () => {
  it("renders app", () => {
    ipc.send = jest.fn();
    jest.spyOn(actions, "setNotificationSystem");
    const component = shallow(<App />);
    expect(toJson(component)).toMatchSnapshot();
    expect(ipc.send).toHaveBeenCalledWith("react-ready");
    expect(actions.setNotificationSystem).toHaveBeenCalled();
  });
});
