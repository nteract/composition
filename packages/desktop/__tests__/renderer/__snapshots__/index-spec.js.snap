// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders app 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <div>
    <Connect(DragDropContext(Notebook))
      displayOrder={
        Array [
          "application/vnd.vega.v2+json",
          "application/vnd.vegalite.v1+json",
          "application/geo+json",
          "application/vnd.plotly.v1+json",
          "text/vnd.plotly.v1+html",
          "application/x-nteract-model-debug+json",
          "application/vnd.dataresource+json",
          "application/vdom.v1+json",
          "application/json",
          "application/javascript",
          "text/html",
          "text/markdown",
          "text/latex",
          "image/svg+xml",
          "image/gif",
          "image/png",
          "image/jpeg",
          "text/plain",
        ]
      }
      transforms={
        Object {
          "application/geo+json": [Function],
          "application/javascript": [Function],
          "application/json": [Function],
          "application/vdom.v1+json": [Function],
          "application/vnd.dataresource+json": [Function],
          "application/vnd.plotly.v1+json": [Function],
          "application/vnd.vega.v2+json": [Function],
          "application/vnd.vegalite.v1+json": [Function],
          "application/x-nteract-model-debug+json": [Function],
          "image/gif": [Function],
          "image/jpeg": [Function],
          "image/png": [Function],
          "image/svg+xml": [Function],
          "text/html": [Function],
          "text/latex": [Function],
          "text/markdown": [Function],
          "text/plain": [Function],
          "text/vnd.plotly.v1+html": [Function],
        }
      }
    />
    <Component
      allowHTML={false}
      noAnimation={false}
      style={Object {}}
    />
  </div>
</Provider>
`;
